go/libraries/doltcore/sqle/enginetest/commit_amend_test.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package enginetest
+
+import (
+	"testing"
+
+	"github.com/dolthub/go-mysql-server/enginetest"
+	"github.com/dolthub/go-mysql-server/enginetest/queries"
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+func TestCommitMessageAmend(t *testing.T) {
+	harness := newDoltHarness(t)
+	enginetest.TestScript(t, harness, queries.ScriptTest{
+		Name: "Sysbench Transactions Shouldn't Cause Constraint Violations",
+		SetUpScript: []string{
+			"SET @@AUTOCOMMIT=0;",
+			"CREATE TABLE test (id INT PRIMARY KEY );",
+			"INSERT INTO test (id) VALUES (1)",
+			"CALL DOLT_ADD('.');",
+			"CALL DOLT_COMMIT('-m', 'original commit message');",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query: "SELECT  message FROM dolt_log;",
+				Expected: []sql.Row{
+					{"original commit message"},
+					{"checkpoint enginetest database mydb"},
+					{"Initialize data repository"},
+				},
+			},
+			{
+				Query:            "CALL DOLT_COMMIT('--amend', '-m', 'modified commit message');",
+				SkipResultsCheck: true,
+			},
+			{
+				Query: "SELECT  message FROM dolt_log;",
+				Expected: []sql.Row{
+					{"modified commit message"},
+					{"checkpoint enginetest database mydb"},
+					{"Initialize data repository"},
+				},
+			},
+		},
+	})
+}
+func TestCommitChangesAmend(t *testing.T) {
+	harness := newDoltHarness(t)
+	enginetest.TestScript(t, harness, queries.ScriptTest{
+		Name: "Sysbench Transactions Shouldn't Cause Constraint Violations",
+		SetUpScript: []string{
+			"SET @@AUTOCOMMIT=0;",
+			"CREATE TABLE test (id INT PRIMARY KEY );",
+			"INSERT INTO test (id) VALUES (1)",
+			"CALL DOLT_ADD('.');",
+			"CALL DOLT_COMMIT('-m', 'original commit message');",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:    "SELECT COUNT(*) FROM dolt_status;",
+				Expected: []sql.Row{{0}},
+			},
+			{
+				Query: "SELECT  message FROM dolt_log;",
+				Expected: []sql.Row{
+					{"original commit message"},
+					{"checkpoint enginetest database mydb"},
+					{"Initialize data repository"},
+				},
+			},
+			{
+				Query:            "INSERT INTO test (id) VALUES (2)",
+				SkipResultsCheck: true,
+			},
+			{
+				Query:    "SELECT COUNT(*) FROM dolt_status;",
+				Expected: []sql.Row{{1}},
+			},
+			{
+				Query:            "CALL DOLT_ADD('.');",
+				SkipResultsCheck: true,
```

Comment:
Try to only use `SkipResultsCheck` when absolutely necessary (e.g. when a commit hash is being returned and we can't reasonably know ahead of time what value to expect). 

For these `CALL` stored procedure calls, since they should return 0 on success, you should be able to expect: 
`[]sql.Row{{0}}` 

For the `Insert into` statements, you should be able to expect something like:
`[]sql.Row{{sql.NewOkResult(1)}},`

Reply:
Excellent point and thanks for the suggestions, removing all the possible `SkipResultsCheck`.